govcd/vm_test.go

```diff
@@ -880,3 +880,40 @@ func (vcd *TestVCD) Test_VMPowerOnPowerOff(check *C) {
 	check.Assert(err, IsNil)
 	check.Assert(vmStatus, Equals, "POWERED_OFF")
 }
+
+// Test gathering VM virtual hardware items
+func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
+	itemName := "TestGetVirtualHardwareSection"
+
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp wasn't properly created")
+	}
+
+	fmt.Printf("Running: %s\n", itemName)
+
+	newVM := types.VM{
+		VirtualHardwareSection: &types.VirtualHardwareSection{
+			Item: []*types.VirtualHardwareItem{
+				&types.VirtualHardwareItem{
+					InstanceID:          1,
+					AutomaticAllocation: true,
+					Address:             "1.1.1.1",
+					CoresPerSocket:      2,
+				},
+			},
+		},
+	}
+
+	vm, err := vcd.client.Client.FindVMByHREF(newVM.HREF)
```

Comment:
This call cannot work.
You are calling a function that finds a VM by HREF, but the HREF is only available for entities that have been created or fetched from vCD.

The test should use an already existing VM (preferred), or create a new one, and then get its contents.